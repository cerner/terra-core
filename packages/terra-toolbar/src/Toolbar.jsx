import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import classNamesBind from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import styles from './Toolbar.module.scss';

const cx = classNamesBind.bind(styles);

const propTypes = {
  /**
   * Content alignment. Align to start, end, or center.
   */
  align: PropTypes.oneOf(['start', 'end', 'center']),
  /**
   * Items to be displayed in toolbar such as buttons, button groups, and links.
   */
  children: PropTypes.node,
};

const defaultProps = {
  align: 'start',
};

const Toolbar = ({
  align, children, ...customProps
}) => {
  const theme = React.useContext(ThemeContext);

  const toolbarClassNames = classNames(
    cx(
      'toolbar',
      `${align}-align`,
      theme.className,
    ),
    customProps.className,
  );

  const items = React.Children.map(children, item => (<div className={cx('item')}>{item}</div>));

  return (<div {...customProps} className={toolbarClassNames}>{items}</div>);
};

Toolbar.propTypes = propTypes;
Toolbar.defaultProps = defaultProps;

export default Toolbar;
