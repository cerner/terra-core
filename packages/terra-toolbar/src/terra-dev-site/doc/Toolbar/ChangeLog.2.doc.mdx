import { Badge } from 'terra-toolbar/package.json?dev-site-package';

import ChangeLog from 'terra-toolbar/CHANGELOG.md';

<Badge />

<ChangeLog />
