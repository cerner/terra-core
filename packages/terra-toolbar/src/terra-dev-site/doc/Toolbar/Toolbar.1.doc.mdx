import { Badge } from 'terra-toolbar/package.json?dev-site-package';
import DefaultToolbar from '../example/DefaultToolbar?dev-site-example';
import PropsTable from 'terra-toolbar/src/Toolbar?dev-site-props-table';

<Badge />

# Terra Toolbar

The terra-toolbar component is used to display bar containing items such as buttons, button groups, and links.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-toolbar`

<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
## Peer Dependencies

This component requires the following peer dependencies be installed in your app for the component to properly function.

| Peer Dependency | Version |
|-|-|
| react | ^16.8.5 |
| react-dom | ^16.8.5 |

<!-- AUTO-GENERATED-CONTENT:END -->

## Usage

```jsx
import Toolbar from 'terra-toolbar';
```

## Component Features

 * [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
 * [Responsive Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#responsive-support)
 * [Mobile Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#mobile-support)
 * [LTR/RTL Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#ltr--rtl-support)

 [View component source code](https://github.com/cerner/terra-core/tree/master/packages/terra-toolbar)

 ## Examples
 <DefaultToolbar />

 ## Toolbar Props
 <PropsTable />
