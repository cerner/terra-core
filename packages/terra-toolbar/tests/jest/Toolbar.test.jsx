import React from 'react';
import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';

import Toolbar from '../../src/Toolbar';

describe('Toolbar', () => {
  const defaultRender = <Toolbar />;

  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('align-start'));
  });

  it('should render with children', () => {
    const wrapper = shallow(<Toolbar><div>test</div></Toolbar>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render start aligned component', () => {
    const wrapper = shallow(<Toolbar align="start" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('align-start'));
  });

  it('should render end aligned component', () => {
    const wrapper = shallow(<Toolbar align="end" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('align-end'));
  });

  it('should render center aligned component', () => {
    const wrapper = shallow(<Toolbar align="center" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('align-center'));
  });

  it('should apply custom class', () => {
    const wrapper = shallow(<Toolbar className="testing" />);
    expect(wrapper.hasClass('testing'));
  });

  it('correctly applies the theme context className', () => {
    const wrapper = mount(
      <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
        {defaultRender}
      </ThemeContextProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
