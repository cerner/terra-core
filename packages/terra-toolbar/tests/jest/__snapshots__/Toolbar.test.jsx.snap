// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar correctly applies the theme context className 1`] = `
<ThemeContextProvider
  theme={
    Object {
      "className": "orion-fusion-theme",
    }
  }
>
  <Toolbar
    align="start"
  >
    <div
      className="toolbar start-align orion-fusion-theme"
    />
  </Toolbar>
</ThemeContextProvider>
`;

exports[`Toolbar should render a default component 1`] = `
<div
  className="toolbar start-align"
/>
`;

exports[`Toolbar should render center aligned component 1`] = `
<div
  className="toolbar center-align"
/>
`;

exports[`Toolbar should render end aligned component 1`] = `
<div
  className="toolbar end-align"
/>
`;

exports[`Toolbar should render start aligned component 1`] = `
<div
  className="toolbar start-align"
/>
`;

exports[`Toolbar should render with children 1`] = `
<div
  className="toolbar start-align"
>
  <div
    className="item"
    key=".0"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
